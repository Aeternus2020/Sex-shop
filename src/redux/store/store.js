import { applyMiddleware, combineReducers, createStore } from "redux";
import products from "../getProducts/reducers";
import basket from "../pages/basket/reducers";
import favorite from "../pages/favorite/reducers";
import showModal from "../components/modal/reducers"; 
import title from "../pages/app/reducers";
import { composeEnhancers, middleware } from "./middleware";

export const rootReducer = combineReducers({
    products,
    basket,
    favorite,
    showModal,
    title
})

export const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(...middleware))
    )