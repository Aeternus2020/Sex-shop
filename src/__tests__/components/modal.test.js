import { render, screen, cleanup } from "@testing-library/react";
import { store } from "../../redux/store/store";
import { Provider } from "react-redux";
import Modal from "../../components/modal";
import renderer from 'react-test-renderer';

afterEach(cleanup);

const item = {
    name: "Whip",
    price: "180$",
    url: "./images/whip.webp",
    article: "52169875",
    color: "black"
}

test('Test basic props in Modal', () => {
    render(
        <Provider store={store} >
            <Modal id={4} item={item}/>
        </Provider>
        );
    expect(screen.getByTestId('modalHtml-4')).toBeInTheDocument();
})

test('Test snapshot Modal', () => {
    const tree = renderer.create(
    <Provider store={store} >
        <Modal id={4} item={item}/>
    </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

