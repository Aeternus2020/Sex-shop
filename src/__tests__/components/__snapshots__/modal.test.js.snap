// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Modal 1`] = `
<div
  className="html"
  data-testid="modalHtml-4"
/>
`;

exports[`snapshot Modal 1`] = `
<div
  className="html"
  data-testid="modalHtml-4"
/>
`;
